spm-integration-test.sh

```diff
@@ -0,0 +1,90 @@
+UUID=$(uuidgen)
+BRANCH_NAME="$1-temp-${UUID}"
+SAMPLE_APP_TEMP_DIR="NativeAuthSampleAppTemp"
+current_date=$(date +"%Y-%m-%d %H:%M:%S")
+
+# Build framework
+
+git checkout -b "$BRANCH_NAME"
+rm -rf archive framework MSAL.zip
+
+xcodebuild -sdk iphonesimulator -configuration Release -workspace MSAL.xcworkspace -scheme "MSAL (iOS Framework)" archive SKIP_INSTALL=NO BUILD_LIBRARY_FOR_DISTRIBUTION=YES -archivePath archive/iOSSimulator CODE_SIGNING_ALLOWED=NO -quiet > build.log 2>&1
+XCBUILD_STATUS_SIM=$?
+if [ $XCBUILD_STATUS_SIM -ne 0 ]; then
+  echo "** Build xcframework Error **"
+  exit 1
+fi
+
+xcodebuild -sdk iphoneos -configuration Release -workspace MSAL.xcworkspace -scheme "MSAL (iOS Framework)" archive SKIP_INSTALL=NO BUILD_LIBRARY_FOR_DISTRIBUTION=YES -archivePath archive/iOS CODE_SIGNING_ALLOWED=NO -quiet > build.log 2>&1
+XCBUILD_STATUS_IPHONE=$?
+if [ $XCBUILD_STATUS_IPHONE -ne 0 ]; then
+  echo "** Build xcframework Error **"
+  exit 1
```

Comment:
should remove the folders created before exit?

Reply:
As long as the machine shuts down after the test it's not necessary as it gets cleaned up, but what needs to happen for sure is that the machine has to shut down. I know there is a way to keep it on so the tests use previous data but that should not be the case for this validation

Reply:
I have updated the job. Now it does the cleanup in a different step after having run the script. The structure is similar to this:

- Step: select Xcode
- Step: Run script, with `continueOnError: true`, so it continues to the next step even if the scrip fails.
- Step: Cleanup

[Here](https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1313778&view=logs&j=1494a728-882c-596b-7b5c-9fa51f14f0ef) there's an example of the cleanup if the SPM scrip fails: